packages/compartment-mapper/src/map-parser.js

```diff
@@ -293,10 +232,49 @@ export const mapParsers = (
     throw Error(`No parser available for language: ${problems.join(', ')}`);
   }
   return makeExtensionParser(
+    preferSynchronous,
     fromEntries(languageForExtensionEntries),
     languageForModuleSpecifier,
     parserForLanguage,
     moduleTransforms,
     syncModuleTransforms,
   );
 };
+
+/**
+ * Prepares a function to map parsers after verifying whether synchronous behavior is 
+ * preferred. Synchronous behavior is selected if all parsers are synchronous and no
+ * async transforms are provided.
+ * 
+ * @param {object} options
+ * @param {ParserForLanguage} options.parserForLanguage
+ * @param {ModuleTransforms} [options.moduleTransforms]
+ * @param {SyncModuleTransforms} [options.syncModuleTransforms]
+ * @returns {(languageForExtension: LanguageForExtension, languageForModuleSpecifier: Record<string, string>) => ParseFnAsync|ParseFn}
```

Comment:
if it were me, I'd create a `MakeMapParsersOptions` type and a `@callback` type for the return value